.github/workflows/ci-build.yml

```diff
@@ -0,0 +1,57 @@
+
+name: CI
+
+on:
+  push:
+  pull_request:
+    branches: [ master ]
+   
+jobs:
+  Code-Quality-Checks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: 
+        scripts:
+          - run-linter
+          - check-docstyle
+          - detect-dead-code
+          - detect-common-errors
+          - measure-cyclomatic-complexity
+          - measure-maintainability-index    
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.6'
+    - name: ${{ matrix.scripts }}
+      run: bash qa/${{ matrix.scripts }}.sh;
+  Unit-Tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.6'
+    - name: Install virtualenv
+      run: pip install -U pip && pip install virtualenv
+    - name: Run Unit Tests
+      run: bash qa/runtests.sh
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        file: ./coverage.xml
+    - name: Install oc
+      uses: redhat-actions/oc-installer@v1
+      with:
+        version: '4.6'
+    - name: Run oc process
+      run: oc process --local -f openshift/template.yaml -o yaml > blueprint.yaml
+    - name: validate openshift yaml
+      uses: instrumenta/kubeval-action@master
+      with:
+          files: blueprint.yaml
```

Comment:
You may need to run this as a separate job.

Reply:
Yup I will make it as a different job

Reply:
@sawood14012 , Also you need to create another job which just builds docker image to ensure Dockerfile sanity.

Reply:
done